adapters/repos/db/migrator.go

```diff
@@ -210,7 +209,8 @@ func (m *Migrator) updateIndexAddMissingProperties(ctx context.Context, idx *Ind
 		// that the property needs to be added to the index, and
 		// don't need to continue iterating over all shards
 		errMissingProp := errors.New("missing prop")
-		err := idx.ForEachShard(func(name string, shard ShardLike) error {
+		// Ensure we iterate over loaded shard to avoid force loading a lazy loaded shard
+		err := idx.ForEachLoadedShard(func(name string, shard ShardLike) error {
```

Comment:
In practice only this fix is necessary to cover the case where we're reloading the DB